File: 3940 limit-occurrences-in-sorted-array/solution.cpp
class Solution {
public:
    vector<int> limitOccurrences(vector<int>& nums, int k) {
        int l=0,c=0;
        for(int i=0;i<nums.size();i++) {
            if(i==0 || nums[i]!=nums[i-1]) {
                c=1;
            } else {
                c++;
            }
            if(c<=k) {
                nums[l++] = nums[i];
            }
        }
        return vector<int>(nums.begin(),nums.begin()+l);
    }
};

File: 3940 limit-occurrences-in-sorted-array/solution.rs
impl Solution {
    pub fn limit_occurrences(mut nums: Vec<i32>, k: i32) -> Vec<i32> {
        let mut l = 0;
    let mut c = 0;

    for i in 0..nums.len() {
        if i == 0 || nums[i] != nums[i - 1] {
            c = 1;
        } else {
            c += 1;
        }
        if c <= k {
            nums[l] = nums[i];
            l += 1;
        }
    }
    nums.truncate(l); 
    nums
    }
}
